### PR TITLE
point filter dialog changes

### PR DESCRIPTION
**Pull request by emre2038**
## Description
* collapsing / expanding the point mapper filter dialog by javascript instead of webflow interactions
* point mapper filter dialog is collapsed by default
* if there are no fields to filter by, we now show a message that's stating no filters available instead of empty dropdowns

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-149

## How is it tested automatically?
added test steps into `__tests__/gui/point_data.feature`

## How should a reviewer test it locally
* select a category from the point mapper
* wait until point markers are created
* confirm that the point mapper filter dialog is collapsed by default
* expand it
* confirm that if there are no fields to filter by, `No filters available for the selected data.` message is shown in the dialog
   -> e.g Higher Education > Universities on YE profile
* confirm that if there are fields to filter by, filtering works as normal
   -> e.g Higher Education > TVET Colleges on YE profile
* confirm that chart filtering, data mapper filtering, point mapper filtering work as expected

## Screenshots
image

### Updated
* `src/js/map/point_data/point_filter.js` to use the new clone of the point filter dialog instead of the old one
* `src/js/map/point_data/filter_controller.js` to show/hide the warning message depending on whether there are fields to filter by or not

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design match the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [x]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

File: src/js/map/point_data/filter_controller.js [hunk 1]
```diff
@@ -145,6 +186,21 @@ export class FilterController extends Component {
 
         this.updateAvailableFiltersOfRows();
         this.model.dataFilterModel.updateFilteredData();
+
+        const addFilterRow = this.noFiltersAvailable && this.model.dataFilterModel.availableFilters.length > 0;   //no filters were available but now there are filterable fields
+        const removeFilterRow = !this.noFiltersAvailable && this.model.dataFilterModel.availableFilters.length <= 0;    //there were filterable fields but now there are none
+
+        this.noFiltersAvailable = this.model.dataFilterModel.availableFilters.length <= 0;
+
+        if (addFilterRow) {
```
Comment (michaelglenister) [inline src/js/map/point_data/filter_controller.js:195]: Making sure I understand correctly. So this.noFiltersAvailable is based on the current data has filters and the extra check is needed if the filters change when selecting another dataset?

**Review by michaelglenister — COMMENTED** [review]
One small change otherwise it looks good

File: src/js/map/point_data/filter_controller.js [hunk 2]
```diff
@@ -91,7 +128,7 @@ export class FilterController extends Component {
             const self = this;
 
             let filterRowContainer = self._rowContainer.cloneNode(true);
-            $(filterRowContainer).show();
+            $(filterRowContainer).removeClass('hidden').show();
```
[the remark this answers is not in the document]
Comment (emre2038) [reply src/js/map/point_data/filter_controller.js:131]: nope @michaelglenister the element had display:none and `hidden` class at the same time. so it was intentional

File: src/js/map/point_data/filter_controller.js (hunk 1 above)
Comment (emre2038) [reply src/js/map/point_data/filter_controller.js:195]: `this.noFiltersAvailable` is assigned `true` only when there are filterable fields

**Pull request merged**